pkg/bff/gds.go

```diff
@@ -155,6 +155,87 @@ func (s *Server) Lookup(c *gin.Context) {
 	c.JSON(http.StatusOK, out)
 }
 
+// VASPNames makes a request on behalf of the user to both the TestNet and MainNet GDS
+// databases and returns the complete deduplicated list of verified VASP names to the
+// user, to facilitate autocomplete functionality in the UI. This is an unauthenticated
+// endpoint so it's publicly accessible.
+//
+// @Summary Get the names of verified VASPs
+// @Description Get the names of all the verified VASPs in both TestNet and MainNet.
+// @Tags GDS
+// @Produce json
+// @Success 200 {list} string "List of VASP names"
+// @Failure 500 {object} api.Reply
+// @Router /vasps [get]
+func (s *Server) VASPNames(c *gin.Context) {
+	// Create an RPC func for making a parallel GDS request for the list of VASPs
+	rpc := func(ctx context.Context, db store.Store, network string) (rep interface{}, err error) {
+		iter := db.ListVASPs(ctx)
+		defer iter.Release()
+
+		var names []string
+		for iter.Next() {
+			var vasp *pb.VASP
+			if vasp, err = iter.VASP(); err != nil {
+				sentry.Error(c).Err(err).Str("network", network).Msg("could not get VASP from ListVASP iterator")
+				return nil, err
+			}
```

Comment:
If we can't parse one VASP record it doesn't make sense to blow up the whole return; instead we'll get the Sentry error and try to fix the corrupted protobuf record specifically for that key:

```suggestion
			if vasp, err = iter.VASP(); err != nil {
				sentry.Error(c).Err(err).Str("network", network).Bytes("key", iter.Key()).Msg("could not get VASP from ListVASP iterator")
				continue
			}
```